### PR TITLE
Qoldev 312 ckan 2.10

### PR DESCRIPTION
**Pull request by ThrawnCA**
- Drop CKAN 2.8 support.
- Increase debugging on test failures.
- Refactor scenario tests to reuse more steps and reduce boilerplate.
- Replace CLI-based stock test data with creation via stable API.
- Handle CKAN version specific plugin dependencies
- Support CKAN 2.10 page header structure

### CODE REVIEW

File: ckanext/data_qld/templates/datarequests/snippets/datarequest_item.html [hunk 1]
```diff
@@ -1,29 +1,7 @@
...
-        {% if h.show_comments_tab() and h.ytp_comments_enabled() %}
-          <a href="{{ h.url_for('datarequest.comment', id=datarequest.get('id','')) }}" class="label label-default"><i class="icon-comment fa fa-comment"></i>{{ h.get_content_type_comments_badge(datarequest.get('id', ''), 'datarequest') }}</a>
-        {% endif %}
-        <div class="divider"/>
-        <span class="date-datarequests">{{ h.time_ago_from_timestamp(datarequest.open_time) }}</span>
-      </div>
-    </div>
-  {% endblock %}
-
+{% block datarequest_comments %}
```
Comment (duttonw) [inline ckanext/data_qld/templates/datarequests/snippets/datarequest_item.html:3]: why such a large change?

File: ckanext/data_qld/templates/package/snippets/package_form.html [hunk 2]
```diff
@@ -1,70 +1,16 @@
...
-                </button>
-            {% endblock %}
-
-            {{ form.required_message() }}
-        </div>
-    {% endblock %}
-</form>
+{% ckan_extends %}
+
+{% block delete_button %}
```
Comment (duttonw) [inline ckanext/data_qld/templates/package/snippets/package_form.html:3]: why such a large change?

Comment (ThrawnCA) [reply ckanext/data_qld/templates/package/snippets/package_form.html:3]: The comments are the only part we actually need to customize. By inheriting the rest, we make our life simpler.

File: ckanext/data_qld/templates/datarequests/snippets/datarequest_item.html (hunk 1 above)
Comment (ThrawnCA) [reply ckanext/data_qld/templates/datarequests/snippets/datarequest_item.html:3]: We don't need to customize anything except the comments.

**Review by duttonw — APPROVED** [review]

**Pull request merged**